Cython/Compiler/Options.py

```diff
@@ -328,6 +330,7 @@ def normalise_encoding_name(option_name, encoding):
     'auto_pickle': ('module', 'cclass'),
     'final' : ('cclass', 'function'),
     'nogil' : ('function', 'with statement'),
+    'device' : ('function'),
```

Comment:
I like that it's the same keyword as used elsewhere for device-offload; it makes it clear this belongs together.
We can also use 'target', but that term is already overloaded in Cython.
I do not like 'omp' because that would make it impossible to write a different offload backend underneath (like based on SYCL) without requiring users to change their code if they want to switch from one to the other.